### DISCUSSION TITLE
VRP or TSP with priortized pickup deliveries

### DISCUSSION THREAD

**Discussion by JohnMartin10000** [google/or-tools#2274] [Routing (and legacy CP) questions]
How can I add priority (ASAP and Normal) for some locations?

It seems that it is not possible for now with Python?

**Comment by lperron** [comment]
This is not an issue. Moving to discussions.

**Comment by Mizux** [comment]
Did you try to use `SetCumulVarSoftUpperBound()` to incentive solver to visit ASAP node earlier ?

ref:
https://github.com/google/or-tools/blob/b77bd3ac69b7f3bb02f55b7bab6cbb4bab3917f2/ortools/constraint_solver/routing.h#L2494-L2503
